### PR TITLE
files-review-p2

### PR DESCRIPTION
**Pull request by coolboy**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @coolboy has signed the CLA. Thanks for the pull request!

File: Box.V2/Managers/BoxFilesManager.cs [hunk 1]
```diff
@@ -141,19 +139,12 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
         /// If the file already exists, an error will be thrown.
         /// A proper timeout should be provided for large uploads.
         /// </summary>
-        /// <param name="fileRequest">Upload file data.
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: what's the correct way?

File: Box.V2/Models/Request/BoxPreflightCheckRequest.cs [hunk 2]
```diff
@@ -8,6 +8,11 @@ namespace Box.V2.Models
 {
     /// <summary>
     /// A request class for making preflight check requests
+    /// 
+    /// Required fields:
+    /// name: The name of the file to be uploaded.
+    /// parent.id: The parent folder id of this file.
```
Comment (coolboy) [inline Box.V2/Models/Request/BoxPreflightCheckRequest.cs]: what's the correct way?

File: Box.V2/Models/Request/BoxPreflightCheckRequest.cs [hunk 3]
```diff
@@ -8,6 +8,11 @@ namespace Box.V2.Models
...
+    /// size: The size of the file in bytes. Specify 0 for unknown file-sizes.
     /// </summary>
     public class BoxPreflightCheckRequest : BoxItemRequest
     {
```
Comment (coolboy) [inline Box.V2/Models/Request/BoxPreflightCheckRequest.cs]: we are missing
1. delete old version
2. promote a version // this is in a PR?

File: Box.V2/Managers/BoxFilesManager.cs [hunk 4]
```diff
@@ -90,9 +90,7 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
         /// folder and file name restrictions*
         /// folder and account storage quota
         /// </remarks>
-        /// <param name="preflightCheckRequest">Fill required inputs: Name - The name of the file to be uploaded, Parent.Id - The ID of the parent folder.,
-        /// Size - The size of the file in bytes. Specify 0 for unknown file-sizes
```
Comment (cburnette) [inline Box.V2/Managers/BoxFilesManager.cs]: what you have looks good

File: Box.V2/Managers/BoxFilesManager.cs (hunk 1 above)
Comment (cburnette) [inline Box.V2/Managers/BoxFilesManager.cs]: what you have is good

File: Box.V2/Managers/BoxFilesManager.cs [hunk 5]
```diff
@@ -281,8 +273,10 @@ private string HexStringFromBytes(byte[] bytes)
         /// and creating a shared link for the file. To move a file, change the ID of its parent folder. An optional etag
         /// can be included to ensure that client only updates the file if it knows about the latest version.
         /// </summary>
-        /// <param name="fileRequest"></param>
-        /// <returns></returns>
+        /// <param name="fileRequest">BoxFileReuqest object.</param>
```
Comment (cburnette) [inline Box.V2/Managers/BoxFilesManager.cs]: typo

File: Box.V2/Managers/BoxFilesManager.cs [hunk 6]
```diff
@@ -281,8 +273,10 @@ private string HexStringFromBytes(byte[] bytes)
...
+        /// <param name="etag">The etag of the file. This is in the ‘etag’ field of the file object.</param>
```
Comment (cburnette) [inline Box.V2/Managers/BoxFilesManager.cs]: need to mention that this is for setting the If-Match Header value

File: Box.V2/Managers/BoxFilesManager.cs [hunk 7]
```diff
@@ -550,8 +544,13 @@ private int GetTimeDelay(HttpResponseHeaders headers)
         /// it was moved to the trash. If that parent folder no longer exists or if there is now an item with the same name in that 
         /// parent folder, the new parent folder and/or new name will need to be included in the request.
         /// </summary>
-        /// <param name="fileRequest">Fill required inputs: Name  - The new name for this item, Id - id of the file. Optional input: Parent - The new parent folder for this item </param>
-        /// <param name="fields">Attribute(s) to include in the response</param>
+        /// <param name="fileRequest">
```
Comment (cburnette) [inline Box.V2/Managers/BoxFilesManager.cs]: just say BoxFileRequest object

File: Box.V2/Managers/BoxFilesManager.cs [hunk 8]
```diff
@@ -606,9 +605,9 @@ private int GetTimeDelay(HttpResponseHeaders headers)
         /// <summary>
         /// Used to update the lock information on the file (for example, ExpiresAt or IsDownloadPrevented.
         /// </summary>
-        /// <param name="lockFileRequest">Request contains Lock object for setting of lock properties such as ExpiresAt - the time the lock expires, IsDownloadPrevented - whether or not the file can be downloaded while locked. </param>
-        /// <param name="id">Id of the file</param>
-        /// <returns>Returns information about locked file</returns>
+        /// <param name="lockFileRequest">BoxFileLockReuqest object.</param>
```
Comment (cburnette) [inline Box.V2/Managers/BoxFilesManager.cs]: typo

File: Box.V2/Models/Request/BoxPreflightCheckRequest.cs (hunk 3 above)
Comment (cburnette) [inline Box.V2/Models/Request/BoxPreflightCheckRequest.cs]: I don't know.  We should look.  If not we need to implement.

File: Box.V2/Managers/BoxFilesManager.cs (hunk 5 above)
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: done

File: Box.V2/Managers/BoxFilesManager.cs (hunk 6 above)
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: done

File: Box.V2/Managers/BoxFilesManager.cs (hunk 7 above)
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: done

File: Box.V2/Managers/BoxFilesManager.cs (hunk 8 above)
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: done

**Pull request merged**